pkg/operator/endpoints/get_batch_job.go

```diff
@@ -48,29 +48,33 @@ func GetJob(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	jobKey := spec.JobKey{APIName: apiName, ID: jobID}
+	jobKey := spec.JobKey{
+		APIName: apiName,
+		ID:      jobID,
+		Kind:    userconfig.BatchAPIKind,
+	}
 
 	jobStatus, err := batchapi.GetJobStatus(jobKey)
 	if err != nil {
 		respondError(w, r, err)
 		return
 	}
 
-	spec, err := operator.DownloadAPISpec(jobStatus.APIName, jobStatus.APIID)
+	apiSpec, err := operator.DownloadAPISpec(jobStatus.APIName, jobStatus.APIID)
 	if err != nil {
 		respondError(w, r, err)
 		return
 	}
 
-	endpoint, err := operator.APIEndpoint(spec)
+	endpoint, err := operator.APIEndpoint(apiSpec)
 	if err != nil {
 		respondError(w, r, err)
 		return
 	}
 
-	response := schema.JobResponse{
+	response := schema.BatchJobResponse{
 		JobStatus: *jobStatus,
-		APISpec:   *spec,
+		APISpec:   *apiSpec,
 		Endpoint:  urls.Join(endpoint, jobKey.ID),
```

Comment:
This endpoint is not correct (it was broken before this PR). I believe the endpoint is `<api_endpoint>/<api_name>?jobID=<job_id>`